File: go/api/api_customer_relations.go
package api

import (
	"github.com/vayucode/vayu-sdks/go/client"
	"github.com/vayucode/vayu-sdks/go/openapi"
)

type CustomerRelation = openapi.CreateCustomerRelationResponseCustomerRelation
type CreateCustomerRelationRequest = openapi.CreateCustomerRelationRequest
type CreateCustomerRelationResponse = openapi.CreateCustomerRelationResponse
type GetCustomerRelationResponse = openapi.GetCustomerRelationResponse

type CustomerRelationsAPI struct {
	vayuClient *client.VayuClient
}

func NewCustomerRelationsAPI(vayuClient *client.VayuClient) *CustomerRelationsAPI {
	return &CustomerRelationsAPI{vayuClient: vayuClient}
}

func NewCreateCustomerRelationRequest(vayuParentCustomerId string, vayuChildCustomerId string) *CreateCustomerRelationRequest {
	return &openapi.CreateCustomerRelationRequest{
		VayuParentCustomerId: vayuParentCustomerId,
		VayuChildCustomerId:  vayuChildCustomerId,
	}
}

func (api *CustomerRelationsAPI) CreateCustomerRelation(payload CreateCustomerRelationRequest) (*CreateCustomerRelationResponse, error) {
	ctx, cancel := client.GenerateContextWithTimeout()
	defer cancel()

	request := api.vayuClient.Client.CustomersAPI.CreateCustomerRelation(ctx)
	request = request.CreateCustomerRelationRequest(payload)
	response, _, err := request.Execute()

	if err != nil {
		return nil, client.BuildVayuErrorFromGenericOpenAPIError(err)
	}

	return response, nil
}

func (api *CustomerRelationsAPI) GetCustomerRelation(customerRelationId string) (*GetCustomerRelationResponse, error) {
	ctx, cancel := client.GenerateContextWithTimeout()
	defer cancel()

	request := api.vayuClient.Client.CustomersAPI.GetCustomerRelation(ctx, customerRelationId)
	response, _, err := request.Execute()

	if err != nil {
		return nil, client.BuildVayuErrorFromGenericOpenAPIError(err)
	}

	return response, nil
}

File: python/vayu_sdk/apis/api_customer_relations.py
from openapi.api.customers_api import CustomersApi
from openapi.models.create_customer_relation_request import CreateCustomerRelationRequest
from openapi.models.create_customer_relation_response import CreateCustomerRelationResponse
from openapi.models.get_customer_relation_response import GetCustomerRelationResponse
from vayu_sdk.clients.vayu_client import VayuClient


class CustomerRelationsAPI:
    __client: CustomersApi = None

    def __init__(self, vayu_client: VayuClient):
        self.__client = CustomersApi(vayu_client.client)

    def create(self, parent_customer_id: str, child_customer_id: str):
        request = CreateCustomerRelationRequest(
            vayuParentCustomerId=parent_customer_id,
            vayuChildCustomerId=child_customer_id,
        )
        return self.__client.create_customer_relation(create_customer_relation_request=request)

    def get(self, customer_relation_id: str):
        return self.__client.get_customer_relation(customer_relation_id=customer_relation_id)
